lib/tests-from-pull-request.js

```diff
@@ -1,32 +1,38 @@
-const fs = require('fs')
-const readline = require('readline')
+module.exports = getTestsFromPR;
 
-module.exports = getTestsFromPR
-
-function getTestsFromPR (context) {
+function getTestsFromPR(context) {
   return Promise.resolve()
-    .then(() => ['Hi Or'])
-  // .then(() => _getTestsFromPR(context))
+    .then(() => _getTestsFromPR(context));
 }
 
-function _getTestsFromPR (context) {
-  const pull = context.issue()
-  const {repos, pullRequests} = context.github
-  // todo: code to get file contect
-  let newTestLines = []
-  let PRfiles
+function _getTestsFromPR(context) {
+  let PRfiles;
+  const pull = context.issue();
+  const {pullRequests} = context.github;
+
   return pullRequests.getFiles(pull)
     .then((res) => res.data)
     .then((files) => PRfiles = files)
-    .then(() => console.log('FILES FROM PR:\n', PRfiles))
+    .then(() => PRfiles.map((file) => file.patch))
+    .then((patches) => patches.join('\n'))
+    .then(_getTestNamesFromPatch);
+}
+
 
-  // let line = ''
-  // if (line.startsWith('+')) {
-  //   line = line.slice(1).trim() // remove the + and spaces
-  //   if (line.trim().startsWith('it(')) {
-  //     newTestLines.push(line)
-  //   }
-  // }
+function _getTestNamesFromPatch(patch) {
+  let newTestLines = [];
+  const parsedPatch = patch.split('\n');
+  parsedPatch.forEach((line) => {
+    if (line.startsWith('+')) {
+      line = line.slice(1).trim(); // remove the + and spaces
+      if (line.trim().startsWith('it(')) {
+        const _testRegex = new RegExp('(?<=\\(\').+?(?=\'\\))', 'g');
```

Comment:
- line 23 can be a const.
- you can remove the `trim` from this `if` statement since you trimed in line 27

- please give a meaningful name to the regex to tell what it does.

- Change the regex itself to a regular expression literal.

As seen on [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions#Creating_a_regular_expression):

>Using a regular expression literal, which consists of a pattern enclosed between slashes, as follows:
>
>```javascript
>var re = /ab+c/;
>```
>Regular expression literals provide compilation of the regular expression when the script is loaded. If the regular expression remains constant, using this can improve performance.
>
>Or calling the constructor function of the RegExp object, as follows:
>
>```javascript
>var re = new RegExp('ab+c');
>```
>Using the constructor function provides runtime compilation of the regular expression. Use the constructor function when you know the regular expression pattern will be changing, or you don't know the pattern and are getting it from another source, such as user input.

Reply:
learning every day